### ISSUE TITLE
Retest and automate complicated Windows VFS tests

### ISSUE THREAD

**Issue by HanaGemela** [owncloud/client#11346]
Repeat the manual tests with proper file structure

Tests with large complicated file structures need to be prioritised for automation.

https://owncloud.testrail.io/index.php?/tests/view/5026

Actions needed

- [x] retest all the test cases manually
- [x] automate if possible
- [ ] Revisit https://owncloud.testrail.io/index.php?/tests/view/5019

### Test cases (from https://github.com/owncloud/QA/issues/789#issuecomment-1706180402)

Files operations 

| Priority | ID  | Test Case | Steps to reproduce | Expected Result | 
| --------- | --- | --------- | ------------------ | --------------- | 
| 4 :heavy_check_mark: | 1  | Rename a file | 1. Rename a file | File has been renamed on the server as well | 
| 4 :heavy_check_mark: | 1 | Delete a file | 1. Delete a file  | File has been deleted on the server as well 
| 4 :heavy_check_mark: | 1  | Edit content of a file | 1. Edit content of a file  | File has been downloaded and edited content has been synced on the server
| 4 :heavy_check_mark: | 1 | Create a new file | 1. Create a new file  | New file has been added and synced but is not virtual https://github.com/owncloud/client/pull/11466
| 4 :heavy_check_mark: | 1 | Copy and paste a file | Copy a virtual file and Paste it at the same location   | File browser asks or creates a copy https://github.com/owncloud/client/pull/11530
| 4 :heavy_check_mark: | 1 | Copy and paste a file | :heavy_check_mark: 1. Copy virtual file and Paste it outside of owncloud sync folder https://github.com/owncloud/client/pull/11466 <br> :heavy_check_mark: 2. Copy virtual file and Paste it inside a folder  https://github.com/owncloud/client/pull/11506 | Placeholder file gets downloaded and it has been copied
| 4 :heavy_check_mark: | 1 | Move a virtual file | Move a virtual file  | Virtual as well as the physical file on the server have been moved https://github.com/owncloud/client/pull/11506

VFS and Selective sync are mutually exclusive

| Priority | ID  | Test Case | Steps to reproduce | Expected Result |
| --------- | --- | --------- | ------------------ | --------------- |
| 4 :heavy_check_mark: | 1 | VFS and Selective sync | 1. Verify when VFS is on, user cannot select folders for selective sync | When VFS is on, user cannot select folders for selective sync https://github.com/owncloud/client/pull/11452
| 4 :heavy_check_mark: | 1 | VFS and Selective sync | 1. Unselect a folder in selective sync and Turn on VFS | Unselected folder is replaced by its placeholder, other folders are physically present https://github.com/owncloud/client/pull/11452
| 4 :heavy_check_mark: | 1 | VFS and Selective sync | 1. Turn off VFS | All files have been replaced by physical files https://github.com/owncloud/client/pull/11452
| 4 :heavy_check_mark:  | 1 | VFS and Selective sync | 1. Switch between VFS and selective sync quickly | It works, no errors https://github.com/owncloud/client/pull/11518 |

**Comment by TheOneRing** [comment]
Blocked by: https://github.com/owncloud/client/issues/9833

**Comment by Salipa-Gurung** [comment]
> * [ ]  Revisit https://owncloud.testrail.io/index.php?/tests/view/5019

This issue is closed as the listed test cases are covered by automated test. Other test cases that can be automated and which are not covered in automated tests will be listed in another new issue.
